OsmAnd/src/net/osmand/plus/dialogs/ConfigureMapMenu.java

```diff
@@ -911,6 +911,7 @@ public void onClick(DialogInterface dialog, int which) {
 					.setCategory(true).setLayout(R.layout.list_group_title_with_switch).createItem());
 			createCustomRenderingProperties(adapter, activity, customRules, app, selectedProfileColor, nightMode, themeRes);
 		}
+		adapter.sortItemsByOrder();
```

Comment:
Make sortItemsByOrder private and remove it from all other places. It must be called from addItem only!